care/facility/api/viewsets/patient.py

```diff
@@ -197,6 +200,30 @@ def filter_list_queryset(self, request, queryset, view):
         return queryset.filter(facility_id__isnull=show_without_facility)
 
 
+class PatientCustomOrderingFilter(BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        ordering = request.query_params.get("ordering", "")
+
+        if ordering == "category_severity" or ordering == "-category_severity":
+            category_ordering = {
+                category: index + 1
+                for index, (category, _) in enumerate(CATEGORY_CHOICES)
+            }
+            when_statements = [
+                When(last_consultation__category=cat, then=order)
+                for cat, order in category_ordering.items()
+            ]
+            queryset = queryset.annotate(
+                category_severity=Case(
+                    *when_statements,
+                    default=(len(category_ordering) + 1),
+                    output_field=models.IntegerField(),
+                )
+            ).order_by(ordering)
```

Comment:
Shouldn't you be ordering based on the new annotated field?

Reply:
@vigneshhari That is being already done, the annotated field is `category_severity` and here `ordering` can be either `category_severity` or `-category_severity`.